src/pages/ReimbursementAccount/ValidationStep.js

```diff
@@ -1,5 +1,181 @@
+import Navigation from '../../libs/Navigation/Navigation';
+import TextInputWithLabel from '../../components/TextInputWithLabel';
+import Text from '../../components/Text';
+import BankAccount from '../../libs/models/BankAccount';
+import CONST from '../../CONST';
+
+const propTypes = {
+    ...withLocalizePropTypes,
+
+    /** Additional data for the account in setup */
+    achData: PropTypes.shape({
+
+        /** Bank account ID of the VBA that we are validating is required */
+        bankAccountID: PropTypes.number.isRequired,
+
+        /** State of bank account */
+        state: PropTypes.string,
+    }).isRequired,
+
+    /** Error message to display to user */
+    error: PropTypes.string,
+
+    /** Disable validation button if max attempts exceeded */
+    maxAttemptsReached: PropTypes.bool,
+};
+
+const defaultProps = {
+    error: '',
+    maxAttemptsReached: false,
+};
+
+class ValidationStep extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.submit = this.submit.bind(this);
+
+        this.state = {
+            amount1: '',
+            amount2: '',
+            amount3: '',
+            error: '',
+        };
+    }
+
+    submit() {
+        const amount1 = this.filterInput(this.state.amount1);
+        const amount2 = this.filterInput(this.state.amount2);
+        const amount3 = this.filterInput(this.state.amount3);
+
+        // If amounts are all non-zeros, submit amounts to API
+        if (amount1 && amount2 && amount3) {
+            const validateCode = [amount1, amount2, amount3].join(',');
+
+            // Make a call to bankAccounts
+            validateBankAccount(this.props.achData.bankAccountID, validateCode);
+            return;
+        }
+
+        // If any values are falsey, indicate to user that inputs are invalid
+        this.setState({error: 'Invalid amounts'});
+    }
+
+    /**
+     *
+     * Filter input for validation amount
+     * Anything that isn't a number is returned as an empty string
+     * Any dollar amount (e.g. 1.12) will be returned as 112
+     *
+     * @param {String} amount field input
+     *
+     * @returns {String}
+     */
+    filterInput(amount) {
+        let value = amount.trim();
+        if (value === '' || !Math.abs(Str.fromUSDToNumber(value)) || _.isNaN(Number(value))) {
+            return '';
+        }
+
+        // If the user enters the values in dollars, convert it to the respective cents amount
+        if (_.contains(value, '.')) {
+            value = Str.fromUSDToNumber(value);
+        }
+
+        return value;
+    }
+
+    render() {
+        let errorMessage = this.state.error ? this.state.error : this.props.error;
+        if (this.props.maxAttemptsReached) {
+            errorMessage = this.props.translate('validationStep.maxAttemptError');
+        }
+
+        const state = this.props.achData.state;
+        const verifyingUrl = `${CONST.CLOUDFRONT_URL}/images/icons/emptystates/emptystate_reviewing.gif`;
```

Comment:
NAB, since this never changes we could set it in the constructor to `this.verifyingUrl`